src/index.ts

```diff
@@ -194,6 +194,14 @@ class DebuggerHandler<
       }
       handler.dispose();
       delete this._handlers[widget.id];
+
+      // clear the model if the handler being removed corresponds
+      // to the current active debug session
+      if (debug.session?.client?.path === client.path) {
+        const model = debug.model as Debugger.Model;
+        model.clear();
```

Comment:
There could be a call to `debug.stop()` instead, so this would also handle the case when the execution has stopped.

But when the notebook is disposed, its kernel (via `client.kernel`) is already set to `null`, which prevents sending the `disconnect` message.

Also, stopping the debugger wouldn't play nice when two notebooks are using the same kernel.